src/config/types.ts

```diff
@@ -96,6 +96,7 @@ export type IsRouteSupportedHandler = (
 export type IsTokenSupportedHandler = (
   token: Token,
   sourceToken?: Token, // The selected source token, if applicable
+  tokenType?: 'source' | 'destination',
```

Comment:
Alternative could be a breaking change and move both sourceToken and tokenType nested inside another param, say context (or better name). Upside is that it makes the api more extensible.

```ts
export type IsTokenSupportedHandler = (
  token: Token,
  context: {
    sourceToken?: Token, // The selected source token, if     applicable
    tokenType?: 'source' | 'destination',
  },
) => boolean;
```